Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportRenderer.cpp

```diff
@@ -64,12 +64,25 @@ namespace EMStudio
         // Link our RPI::Scene to the AzFramework::Scene
         m_frameworkScene->SetSubsystem(m_scene);
 
-        // Create a render pipeline from the specified asset for the window context and add the pipeline to the scene
-        AZStd::string defaultPipelineAssetPath = "passes/MainRenderPipeline.azasset";
-        AZ::Data::Asset<AZ::RPI::AnyAsset> pipelineAsset = AZ::RPI::AssetUtils::LoadAssetByProductPath<AZ::RPI::AnyAsset>(
-            defaultPipelineAssetPath.c_str(), AZ::RPI::AssetUtils::TraceLevel::Error);
-        m_renderPipeline = AZ::RPI::RenderPipeline::CreateRenderPipelineForWindow(pipelineAsset, *m_windowContext.get());
-        pipelineAsset.Release();
+        const AZ::RPI::RenderPipelineDescriptor renderPipelineDesc =
+            [](const AzFramework::ViewportId& viewportId)
+            {
+                // Load the render pipeline asset
+                const char* pipelineAssetPath = "passes/MainRenderPipeline.azasset";
+                AZ::Data::Asset<AZ::RPI::AnyAsset> pipelineAsset = AZ::RPI::AssetUtils::LoadAssetByProductPath<AZ::RPI::AnyAsset>(
+                    pipelineAssetPath, AZ::RPI::AssetUtils::TraceLevel::Error);
+                const AZ::RPI::RenderPipelineDescriptor* assetPipelineDesc = AZ::RPI::GetDataFromAnyAsset<AZ::RPI::RenderPipelineDescriptor>(pipelineAsset);
+                AZ_Assert(assetPipelineDesc, "Invalid render pipeline descriptor from asset %s", pipelineAssetPath);
+
+                // Use a unique render pipeline name to not conflict with other render pipelines
+                AZ::RPI::RenderPipelineDescriptor pipelineDesc = *assetPipelineDesc;
+                pipelineDesc.m_name += AZStd::string::format("_%i", viewportId);
+
+                pipelineAsset.Release();
+                return pipelineDesc;
+            }(viewportContext->GetId());
+        
```

Comment:
What is the advantage of creating a lambda function and then calling it inline?

Reply:
`renderPipelineDesc` can be `const` and it's more encapsulated as an operation, easier to extract to a function in the future if necessary.

Reply:
I also think the Lambda is unnecessary. You could just copy the pipeline descriptor directly from the asset and append a unique suffix to the descriptor name. Instead of the lamda, this would make a good utility function if this code is repeated in multiple places.

Reply:
I'll try to find a place to put it as an util function.

Reply:
I'll do this improvement in a follow up PR.

Reply:
Follow up PR putting code in an util function: https://github.com/o3de/o3de/pull/7880